src/main/java/iudx/catalogue/apiserver/APIServerVerticle.java

```diff
@@ -445,12 +445,14 @@ private void getConfig(RoutingContext routingContext){
 
   private void list(RoutingContext routingContext) {
     String currentType = routingContext.request().getParam("itemtype");
-
+    host = routingContext.request().getHeader("Host");
+    logger.info("----HIT LIST API-----");
     /**
      * New code-
      * Added as a part to provide List of Tags, Provider, ResourceServerGroup APIs for UI
      **/
     JsonObject requestBody = new JsonObject();
+    requestBody.put("__instance-id",host);
```

Comment:
"host" can also be null !

Reply:
routingContext.request().getHeader("Host") will either give  domain name or the ip:port of the server (as used by the client in the request) always and can never be null. 

> "host" can also be null !

routingContext.request().getHeader("Host") will either give  domain name or the ip:port of the server (as used by the client in the request) always and can never be null.